streams/src/main/java/org/apache/kafka/streams/kstream/internals/suppress/KTableSuppressProcessor.java

```diff
@@ -76,9 +63,8 @@ public void init(final ProcessorContext context) {
         internalProcessorContext = (InternalProcessorContext) context;
         suppressionEmitSensor = Sensors.suppressionEmitSensor(internalProcessorContext);
 
-        keySerde = keySerde == null ? (Serde<K>) context.keySerde() : keySerde;
-        valueSerde = valueSerde == null ? FullChangeSerde.castOrWrap(context.valueSerde()) : valueSerde;
-        buffer = Objects.requireNonNull((TimeOrderedKeyValueBuffer) context.getStateStore(storeName));
+        buffer = requireNonNull((TimeOrderedKeyValueBuffer<K, Change<V>>) context.getStateStore(storeName));
+        buffer.setSerdesIfNull((Serde<K>) context.keySerde(), FullChangeSerde.castOrWrap(context.valueSerde()));
```

Comment:
This is the only potentially weird thing... Rather than the store setting its own serdes during `store init` from the context, I'm letting the processor set them during `processor init`.

The reason for this is that the store is generically typed, and therefore does not know that it needs to wrap the value serde in a Change serde. We could tell it that its value is a Change, but it would be unnatural to define the buffer as a `Store<K, Change<V>>`.

Thus, we give the responsibility to the Processor during initialization, after it resolves the store, to make sure that the serdes are set before using it. This is more natural, since the Processor already knows that the store's value is a Change, so it can confidently wrap the context's serde and hand it off to the store.

It's also safe, because we know that no other caller will interact with the store's typed methods except the Processor (e.g., the store is (and will always be) inaccessible to IQ).

Reply:
> but it would be unnatural to define the buffer as a Store<K, Change<V>>.

Not sure if I understand this argument. Can you elaborate?

Reply:
Sure; right now, the buffer is a generic k/v store with special time-dependent semantics. There's nothing in the buffer implementation that requires the value to be a specific type.

At the same time, we have to deal with this whole "try harder to set the serdes if they didn't get set to begin with" business. The suppress operator (not the buffer) happens to require a very specific serde, the `FullChangeSerde`, which is actually an internal class, so we're basically guaranteed to get the wrong serde and get a runtime exception if we just try to use whatever serde is defined in the context. But this isn't the buffer's problem, it's the operator's problem. So, that's why I chose to add this method to let the operator deal with its own business and keep the buffer ignorant of the specific value type.

Reply:
While I was wrestling with this, I got some inspiration that can actually let us get rid of these runtime serde double-checks everywhere, but it's out of scope for this PR. Hopefully, we can remove this method later on, though, and just guarantee that the serdes are never null in all stores and operators.

Reply:
Thanks!